annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/attribute/constructor/ItemConstructor.kt

```diff
@@ -0,0 +1,71 @@
+package com.tickaroo.tikxml.annotationprocessing.attribute.constructor
+
+import com.tickaroo.tikxml.annotation.Attribute
+import com.tickaroo.tikxml.annotation.Xml
+import com.tickaroo.tikxml.annotationprocessing.DateConverter
+import java.util.Date
+
+/**
+ * @author Hannes Dorfmann
+ */
+@Xml(name = "item")
+class ItemConstructor(
```

Comment:
Nice!
But we need the same class (and test) for data classes `data class ItemConstructorDataClass( @Attribute val aString : String, ... )`

Reply:
Ok, I will add data classes and tests for them.
Is it necessary to do all of them or small portion?

Reply:
> Is it necessary to do all of them or small portion?

I dont understand the question.
Do you mean all attributes? Then yes.

Reply:
I meant for all classes create equivalent data class (if it makes sense in that context).
I think that data class is not only constructor it will generate getters/setters as well.
I will take a look how `@param:Annotation` and `@Annotation` affect code generation in data classes and go from there.

Reply:
Got it, yes I think that makes sense to have for all constructor class a corresponding data class (because under the hood those are 2 entirely different code paths), but if it's too much work for now I can add those data classes and tests later.

Reply:
Looks like behavior is same as normal class.
If I use simple `@Annotation` it will show up only on field declaration and with `@param:` it will show up in constructor declaration.
It is possible to add `@JvmField` as well.

Reply:
Oh, I was wrong with annotations..
I am looking at decompiled source once again and `@Annotation` without prefix `@param:` will not be in field declaration. (was testing with `@JvmField` wrong idea).